python/src/scipp/plot/plot_3d.py

```diff
@@ -82,292 +87,546 @@ def __init__(self,
+        elif self.cut_surface_buttons.value < 6:
+            j = self.cut_surface_buttons.value - 3
+            remaining_axes = self.permutations["xyz"[j]]
+            self.remaining_inds = [(j + 1) % 3, (j + 2) % 3]
+            rmax = np.abs([
+                self.xminmax[remaining_axes[0]][0],
+                self.xminmax[remaining_axes[1]][0],
+                self.xminmax[remaining_axes[0]][1],
+                self.xminmax[remaining_axes[1]][1]
+            ]).max()
+            self.cut_slider.min = 0
+            self.cut_slider.max = rmax * np.sqrt(2.0)
+            self.cut_slider.value = 0.5 * self.cut_slider.max
+        # Spherical
+        elif self.cut_surface_buttons.value == 6:
+            rmax = np.abs(list(self.xminmax.values())).max()
+            self.cut_slider.min = 0
+            self.cut_slider.max = rmax * np.sqrt(3.0)
+            self.cut_slider.value = 0.5 * self.cut_slider.max
+        # Value iso-surface
+        elif self.cut_surface_buttons.value == 7:
+            self.cut_slider.min = self.vminmax[0]
+            self.cut_slider.max = self.vminmax[1]
+            self.cut_slider.value = 0.5 * (self.vminmax[0] + self.vminmax[1])
+
+        # Remember to reset surface thickness when changing between spatial
+        # slicing and value slicing
+        if self.cut_surface_buttons.value < 7:
+            self.cut_surface_thickness.value = 0.05 * self.box_size.max()
+        else:
+            self.cut_surface_thickness.value = 0.05 * (self.vminmax[1] -
+                                                       self.vminmax[0])
+
+        # Update slider step to avoid too fine granularity which slows down
+        # interaction.
+        self.cut_slider.step = (self.cut_slider.max -
+                                self.cut_slider.min) / self.cut_slider_steps
+
+    def update_cut_surface(self, change):
+        newc = None
+        # Cartesian X, Y, Z
+        if self.cut_surface_buttons.value < 3:
+            newc = np.where(
+                np.abs(self.positions[:, self.cut_surface_buttons.value] -
+                       change["new"]) < self.cut_surface_thickness.value,
+                self.opacity_slider.upper, self.opacity_slider.lower)
+        # Cylindrical X, Y, Z
+        elif self.cut_surface_buttons.value < 6:
+            newc = np.where(
+                np.abs(
+                    np.sqrt(self.positions[:, self.remaining_inds[0]] *
+                            self.positions[:, self.remaining_inds[0]] +
+                            self.positions[:, self.remaining_inds[1]] *
+                            self.positions[:, self.remaining_inds[1]]) -
+                    change["new"]) < self.cut_surface_thickness.value,
+                self.opacity_slider.upper, self.opacity_slider.lower)
+        # Spherical
+        elif self.cut_surface_buttons.value == 6:
+            newc = np.where(
+                np.abs(
+                    np.sqrt(self.positions[:, 0] * self.positions[:, 0] +
```

Comment:
I think for this to be useful you want to shift into the scattering frame of reference. This should be the case if the sample is at the origin (nexus requirement, but not always the case in mantid).

Reply:
Yes, i guess for this and the cylinder to be useful one should be able to change the origin. But I suggest we add this later, once someone who is actually using it needs this extra feature.